src/services/services.ts

```diff
@@ -1278,11 +1278,10 @@ namespace ts {
             }
 
             function getOrCreateSourceFileByPath(fileName: string, path: Path, _languageVersion: ScriptTarget, _onError?: (message: string) => void, shouldCreateNewSourceFile?: boolean): SourceFile | undefined {
-                Debug.assert(hostCache !== undefined);

```

Comment:
I think you do want to assert that hostCache !== undefined. That is getOrCreateSourceFileByPath should be called only when hostCache is not undefined.